tservice/src/report.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2022 The MobileCoin Foundation
+//! Functions used for creating and verifying reports inside of an enclave
+
+use core::ptr;
+use mc_sgx_core_sys_types::sgx_report_t;
+use mc_sgx_core_types::{ReportData, Result, TargetInfo};
+use mc_sgx_util::ResultInto;
+
+/// Report operations that can be performed inside of an enclave
+pub trait Report {
+    /// Creates a report for the current enclave.
+    ///
+    /// # Arguments
+    /// * `target_info` - The information for the target enclave which will
+    ///   cryptographically verify the report.  See [`Report::verify_report()`].
+    /// * `report_data` - Report data used to communicate between enclaves
+    fn create_report(
```

Comment:
nit: Recommend just `Report::new()` here.

Reply:
Converted to new